lib/ansible/plugins/connections/docker.py

```diff
@@ -0,0 +1,119 @@
+# Based on the chroot connection plugin by Maykel Moya
+#
+# Connection plugin for configuring docker containers
+# (c) 2014, Lorin Hochstein
+# (c) 2015, Leendert Brouwer
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+import os
+import subprocess
+import time
+
+from ansible import errors
+from ansible.plugins.connections import ConnectionBase
+
+BUFSIZE = 65536
+
+class Connection(ConnectionBase):
+
+    def __init__(self, play_context, new_stdin, *args, **kwargs):
+        super(Connection, self).__init__(play_context, new_stdin, *args, **kwargs)
+
+        if 'docker_command' in kwargs:
+            self.docker_cmd = kwargs['docker_command']
+        else:
+            self.docker_cmd = 'docker'
+
+    @property
+    def transport(self):
+        return 'docker'
+
+    def _connect(self, port=None):
+        """ Connect to the container. Nothing to do """
+        return self
+
```

Comment:
I would do something like local.py  does here:
``` python
        if not self._connected:
            self._display.vvv("ESTABLISH DOCKER CONNECTION FOR USER: {0}".format(self._play_context.remote_user,      host=self._play_context.remote_addr))
            self._connected = True
        return self
```

It adds logging which is a good thing to have when things go wrong.  Setting self._connected is unnecessary for docker but it is nice if someone later decides to cargo-cult from the docker module and their connection plugin does need to set the _connected attribute.